core/metadata.go

```diff
@@ -41,6 +41,12 @@ func (meta Metadata) Delete(key string) {
 	delete(meta, key)
 }
 
+// HasValue returns true if the given key exists
+func (meta Metadata) HasValue(key string) bool {
```

Comment:
this is a "KeyExists" method - "HasValue" associates that the value is checked also.